docs/src/upgrade.md

```diff
@@ -0,0 +1,16 @@
+# Upgrading from Legolas v0.4 to v0.5
+
+This guide is incomplete; please add to it if you encounter items which would help other upgraders along their journey.
+
+## Some main changes to be aware of
+
+* In Legolas v0.4, type parameters were implicit in the row types; for example, a column `y::Real` in a `Legolas.@row` declaration would be represented by `y::T` with the constraint `T<:Real`, rather than using the abstract type `Real` as a type declaration for `y`. Now, all type parameters need to manually specified with the `y::(<:Real)` syntax in [`@version`](@ref).
+    * Additionally, to include type parameters associated to fields in a parent schema, they must be re-declared in the child schema. For example, the package LegolasFlux declares a `ModelV1` version with a field `weights::(<:Union{Missing,Weights})`. LegolasFlux includes an [example](https://github.com/beacon-biosignals/LegolasFlux.jl/blob/53c677848c6b65e5158ef2d43dd5f7eab174892e/examples/digits.jl#L78-L80) with a schema extension `DigitsRowV1` which extends `ModelV1`. This `@version` call must re-declare the field `weights` to be parametric in order for the `DigitsRowV1` struct to also have a type parameter for this field.
+* Constructors for structs generated by `@version` calls discard "extra" columns, whereas previously under Legolas v0.4, `@row`-generated constructors kept any "extra" columns. It may be advisable to add previously used "extra" columns into the schema itself (probably declared as `::Union{Missing,T}` to allow them to be missing).
+
+
+## Deserializing old tables with Legolas v0.5
+
+Generally, tables serialized with Legolas v0.4 can be de-serialized with Legolas v0.5, making it only a "code-breaking" change, rather than a "data-breaking" change. However, it is strongly suggested to have reference tests with checked in (pre-Legolas v0.5) serialized tables which are deserialized and verified during the tests, in order to be sure.
```

Comment:
```suggestion
Generally, tables serialized with earlier versions of Legolas can be de-serialized with Legolas v0.5, making it only a "code-breaking" change, rather than a "data-breaking" change. However, it is strongly suggested to have reference tests with checked in (pre-Legolas v0.5) serialized tables which are deserialized and verified during the tests, in order to be sure.
```